tests/app/views/test_drafts.py

```diff
@@ -0,0 +1,336 @@
+            content_type='application/json')
+        assert_equal(fetch_again.status_code, 404)
+
+    def test_should_be_able_to_update_a_draft(self):
+        self.client.put(
+            '/services/{}/draft'.format(self.service_id),
+            data=json.dumps(self.updater_json),
+            content_type='application/json')
+
+        update = self.client.post(
+            '/services/{}/draft'.format(self.service_id),
+            data=json.dumps({
+                'update_details': {
+                    'updated_by': 'joeblogs',
+                    'update_reason': 'whateves'},
+                'services': {
+                    'serviceName': 'new service name'
+                }
+            }),
+            content_type='application/json')
+
+        data = json.loads(update.get_data())
+        assert_equal(update.status_code, 200)
+        assert_equal(data['services']['serviceName'], 'new service name')
+
+        fetch = self.client.get('/services/{}/draft'.format(self.service_id))
+        data = json.loads(fetch.get_data())
+        assert_equal(fetch.status_code, 200)
+        assert_equal(data['services']['serviceName'], 'new service name')
+
+    def test_should_be_a_400_if_no_service_block_in_update(self):
+        self.client.put(
+            '/services/{}/draft'.format(self.service_id),
+            data=json.dumps(self.updater_json),
+            content_type='application/json')
+
+        update = self.client.post(
+            '/services/{}/draft'.format(self.service_id),
+            data=json.dumps({
+                'update_details': {
+                    'updated_by': 'joeblogs',
+                    'update_reason': 'whateves'}
+            }),
+            content_type='application/json')
+
+        assert_equal(update.status_code, 400)
+
+    def test_should_not_be_able_to_launch_invalid_service(self):
+        self.client.put(
+            '/services/{}/draft'.format(self.service_id),
+            data=json.dumps(self.updater_json),
+            content_type='application/json')
+
+        self.client.post(
+            '/services/{}/draft'.format(self.service_id),
+            data=json.dumps({
+                'update_details': {
+                    'updated_by': 'joeblogs',
+                    'update_reason': 'whateves'},
+                'services': {
+                    'badField': 'new service name',
+                    'priceUnit': 'chickens'
+                }
+            }),
+            content_type='application/json')
+
+        res = self.client.post(
+            '/services/{}/draft/publish'.format(self.service_id),
+            data=json.dumps({
+                'update_details': {
+                    'updated_by': 'joeblogs',
+                    'update_reason': 'whateves'
+                }
+            }),
+            content_type='application/json')
+
+        assert_equal(res.status_code, 400)
+
+    def test_should_not_be_able_to_a_draft_if_service_doesnt_exist(self):
```

Comment:
Should not be able to do what to a draft service? Go over the line length limit?